api/envoy/extensions/filters/http/ext_proc/v3/processing_mode.proto

```diff
@@ -35,6 +35,18 @@ message ProcessingMode {
   }
 
   // Control how the request and response bodies are handled
+  // When body mutation by external processor is enabled, ext_proc filter will always remove
+  // the content length header in 3 cases below:
+  // 1) STREAMED BodySendMode
+  // 2) BUFFERED_PARTIAL BodySendMode
+  // 3) BUFFERED BodySendMode + SKIP HeaderSendMode
+  // In these modes, ext_proc filter can not guarantee to set the content length correctly when
+  // body is mutated by external processor later. In http1 codec, removing content length will
+  // enable chunked encoding whenever feasible.
+  // In BUFFERED BodySendMode + SEND HeaderSendMode, content length header is allowed but it is
+  // external processor's responsibility to set the content length correctly matched to the length
+  // of mutated body. If they don't match, the corresponding body mutation will be rejected and
+  // local reply will be sent with an error message.
```

Comment:
Is this a breaking change for any existing user?

Reply:
Yea it could break users who set the content-length that doesn't match the actual body. But this change is fixing the vulnerability concern (i.e., request smuggling) and we are only breaking the inappropriate use of this feature. 
Therefore, we should be fine here?

Reply:
OK, I think I'm persuaded that this should not have significant impact.